src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RatingControlTests/RatingControlPage.xaml.cs

```diff
@@ -0,0 +1,639 @@
+				Compositor compositor = visualPPChild.Compositor;
+
+				this.AnimatedValuesSpy = compositor.CreatePropertySet();
+
+				this.AnimatedValuesSpy.InsertVector3("Scale", new Vector3(0.0f, 0.0f, 0.0f));
+
+				this.ScaleAnimation = compositor.CreateExpressionAnimation(visualScaleTargetedPropertyName);
+				this.ScaleAnimation.SetReferenceParameter("visual", visualPPChild);
+
+				CheckSpyingTicksRequirement();
+
+				TickForValuesSpy();
+			}
+			else
+			{
+				ResetSpyOutput();
+			}
+		}
+
+		private void StartAnimatedValuesSpy()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				this.AnimatedValuesSpy.StartAnimation("Scale", this.ScaleAnimation);
+			}
+		}
+
+		private void StopAnimatedValuesSpy()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				this.AnimatedValuesSpy.StopAnimation("Scale");
+			}
+		}
+
+		private void SpyAnimatedValues()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				StopAnimatedValuesSpy();
+
+				Vector3 scale;
+				CompositionGetValueStatus status = this.AnimatedValuesSpy.TryGetVector3("Scale", out scale);
+				if (CompositionGetValueStatus.Succeeded == status)
+				{
+					this.ScaleTextX.Text = scale.X.ToString();
+					this.ScaleTextY.Text = scale.Y.ToString();
+				}
+				else
+				{
+					this.ScaleTextX.Text = this.ScaleTextY.Text = "status=" + status.ToString();
+				}
+
+				StartAnimatedValuesSpy();
+
+				// System.Diagnostics.Debug.WriteLine("Spied values: " + this.ScaleTextX.Text);
+			}
+		}
+
+		private void ResetSpyOutput()
+		{
+			this.ScaleTextX.Text = this.ScaleTextY.Text = "0 - reset";
+		}
+
+		private void TickForValuesSpy()
+		{
+			this.UIThreadTicksForValuesSpy = 6;
+			CheckSpyingTicksRequirement();
+		}
+
+		private void CheckSpyingTicksRequirement()
+		{
+			if (this._secondTextBlockUI != null &&
+				(this.UIThreadTicksForValuesSpy > 0) && this.AnimatedValuesSpy != null)
+			{
+				if (!this.IsRenderingHooked)
+				{
+					this.IsRenderingHooked = true;
+					Windows.UI.Xaml.Media.CompositionTarget.Rendering += CompositionTarget_Rendering;
+				}
+			}
+			else
+			{
+				if (this.IsRenderingHooked)
+				{
+					Windows.UI.Xaml.Media.CompositionTarget.Rendering -= CompositionTarget_Rendering;
+					this.IsRenderingHooked = false;
+				}
+			}
+		}
+
+		private void CompositionTarget_Rendering(object sender, object e)
+		{
+			if (this.UIThreadTicksForValuesSpy > 0)
+			{
+				this.UIThreadTicksForValuesSpy--;
+			}
+			CheckSpyingTicksRequirement();
+			SpyAnimatedValues();
+		}
+
+		void PerformGlyphVerification(RatingControl rc, String expectedGlyph, TextBlock resultTb, String stackPanelName, SolidColorBrush expectedColor = null)
+		{
+			ContentPresenter presenter = FindVisualChildByName(rc, "ForegroundContentPresenter") as ContentPresenter;
+
+			DependencyObject obj = FindVisualChildByName(rc, stackPanelName);
+			var child = VisualTreeHelper.GetChild(obj, 0);
+			TextBlock tb = child as TextBlock;
+
+			String glyph = tb.Text;
+			if (glyph.Equals(expectedGlyph))
```

Comment:
Codacy found an issue: [Change this call to 'glyph.Equals' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)